enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,552 @@
+2. Wait for some customers to update to the OCP version (e.g. 500 customer clusters). 
+   This could take a month or two.
+3. Experiment with log requests in small steps and monitor how the feature performs. 
+   The feature enables fast feedback loop within a day. 
+   1. The inclusion of the Insights Operator version into the remote configuration request 
+      also enables version-based canary rollouts. 
+   For instance, we will be able to test a change on pre-release (i.e. mostly internal) clusters first.
+4. Open the feature to data requesters when we are confident that the feature 
+   and our monitoring capabilities are robust enough.
+   This could be aligned with y+1 or y+2 OCP version.
+5. Continue monitoring the feature as more and more customers upgrade to OCP versions with the feature.
+   Until we start replacing existing gatherers by remote configuration, 
+   we will be able to workaround any issues by serving empty/reduced remote configuration 
+   which will affect only new applications specifically depending on the feature.
+
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+N/A
+
+### Removing a deprecated feature
+
+In fact, this is not a removal, but the remote configuration service endpoint is already 
+part of the Insights Operator configuration. 
+This means that the cluster administrator can override this endpoint 
+(or simply insert an empty string) to disable this feature. 
+The obvious consequence for us is that some data from this cluster will be missing, 
+but this information must be visible in the Insights archive 
+(as mentioned in the [Risk and mitigations](#risks-and-mitigations) section).
+
+## Upgrade / Downgrade Strategy
+
+When a cluster version is upgraded or downgraded, Insights Operator requests a new version 
+of the remote configuration (the operator knows and uses the OCP cluster version). 
+As mentioned in the 
+[Remote configuration service](#remote-configuration-service) section, 
+it is the responsibility of the administrator/owner of the remote configuration to provide valid 
+and updated corresponding configuration. Other than that, no special upgrade or downgrade strategy 
+should be needed. 
+We do not expect any changes to the API or the cluster in general. 
+There will be changes to the Insights Operator, but the operator must not be marked as Degraded 
+if the connection to the remote configuration fails 
+(i.e the particular data gathering will be skipped and this information 
+will be available in the Insights archive).
+
+## Version Skew Strategy
+
+This enhancement proposal considers only two components (which are not critical to the cluster). 
+It is the Insights Operator, which is versioned according to the OCP version and the remote configuration,
+which is versioned in the same way. 
+There should not be any version skew between the two 
+(you can see the [Remote configuration service](#remote-configuration-service) section).
```

Comment:
As I mention [here][1], there is still a possibility of mid-update skew between the version of the current Insights operator pod and its associated rules, and the rest of the cluster.  Kube API and etcd compononts might be running one of many later versions.  Network, DNS, and machine-config might be running one of many earlier versions.  I think it's worth calling that out in this version-skew section, but you can probably then wave away the problem by saying that some rules failing to find the information they were hoping to find is an existing problem with hard-coded-for-the-Insights-operator rules, and if some gathering rules miss while a cluster is mid-update, that's fine, because the Insights operator's 2h poll makes it too high-latency to be particularly relevent during a quickly-moving update anyway.  It might return to relevance during an update that wedges for an extended period of time, and maybe that's an interesting place to dig into the rules-vs.-some-cluster-components skew?

[1]: https://github.com/openshift/enhancements/pull/1569#discussion_r1522559209

Reply:
Yes I think it's an acceptable state that we may be missing some data in this case, but I think it's worth mentioning.